File: Rust/S0010-regular-expression-matching.rs
// Given an input string (s) and a pattern (p), implement regular expression matching with
// support for '.' and '*'.
// '.' Matches any single character.
// '*' Matches zero or more of the preceding element.
// The matching should cover the entire input string (not partial).
//
// Note:
// s could be empty and contains only lowercase letters a-z.
// p could be empty and contains only lowercase letters a-z, and characters like . or *.
//
// Example 1:
// Input:
// s = "aa"
// p = "a"
// Output: false
// Explanation: "a" does not match the entire string "aa".
//
// Example 2:
// Input:
// s = "aa"
// p = "a*"
// Output: true
// Explanation: '*' means zero or more of the preceding element, 'a'. Therefore,
// by repeating 'a' once, it becomes "aa".
//
//
// Example 3:
// Input:
// s = "ab"
// p = ".*"
// Output: true
// Explanation: ".*" means "zero or more (*) of any character (.)".
//
//
// Example 4:
// Input:
// s = "aab"
// p = "c*a*b"
// Output: true
// Explanation: c can be repeated 0 times, a can be repeated 1 time. Therefore, it matches "aab".
//
//
// Example 5:
// Input:
// s = "mississippi"
// p = "mis*is*p*."
// Output: false
pub struct Solution {}

impl Solution {
    pub fn is_match(s: String, p: String) -> bool {

    }
}

fn main() {
	assert_eq!(true, Solution::my_atoi("aa".to_string(), "a".to_string()));
    assert_eq!(true, Solution::my_atoi("aa".to_string(), "a*".to_string()));
    assert_eq!(true, Solution::my_atoi("ab".to_string(), ".*".to_string()));
    assert_eq!(true, Solution::my_atoi("az".to_string(), "a.".to_string()));
    assert_eq!(false, Solution::my_atoi("az".to_string(), "a*".to_string()));
    assert_eq!(true, Solution::my_atoi("".to_string(), "b*".to_string()));
    assert_eq!(false, Solution::my_atoi("aaa".to_string(), "".to_string()));
    assert_eq!(true, Solution::my_atoi("".to_string(), "".to_string()));
    assert_eq!(true, Solution::my_atoi("aab".to_string(), "c*a*b*".to_string()));
    assert_eq!(false, Solution::my_atoi("mississippi".to_string(), "mis*is*p*.".to_string()));
    assert_eq!(false, Solution::my_atoi("mississippi".to_string(), "mis*is*ip*.".to_string()));
	println!("Pass test cases!");
}

File: src-all/Python/S0010-regular-expression-matching.py
"""
Given an input string (s) and a pattern (p), implement regular expression matching with support for &#39;.&#39; and &#39;*&#39;.

&#39;.&#39; Matches any single character.
&#39;*&#39; Matches zero or more of the preceding element.

The matching should cover the entire input string (not partial).

Note:
	s could be empty and contains only lowercase letters a-z.
	p could be empty and contains only lowercase letters a-z, and characters like . or *.

Example 1:
Input:
s = &quot;aa&quot;
p = &quot;a&quot;
Output: false
Explanation: &quot;a&quot; does not match the entire string &quot;aa&quot;.

Example 2:
Input:
s = &quot;aa&quot;
p = &quot;a*&quot;
Output: true
Explanation: &#39;*&#39; means zero or more of the preceding element, &#39;a&#39;. Therefore, by repeating &#39;a&#39; once, it becomes &quot;aa&quot;.

Example 3:
Input:
s = &quot;ab&quot;
p = &quot;.*&quot;
Output: true
Explanation: &quot;.*&quot; means &quot;zero or more (*) of any character (.)&quot;.

Example 4:
Input:
s = &quot;aab&quot;
p = &quot;c*a*b&quot;
Output: true
Explanation: c can be repeated 0 times, a can be repeated 1 time. Therefore, it matches &quot;aab&quot;.

Example 5:
Input:
s = &quot;mississippi&quot;
p = &quot;mis*is*p*.&quot;
Output: false


"""
class Solution:
    def isMatch(self, s: str, p: str) -> bool:
        
        pass


if __name__ == '__main__':
    assert Solution().isMatch(0) == 0

